components/datetime/src/time_zone.rs

```diff
@@ -176,6 +175,8 @@ impl<'data> TimeZoneFormat<'data> {
 
         let zone_symbols = time_zone_format
             .patterns
+            .as_ref()
+            .expect("TimeZoneFormat::try_new() expected Some(patterns) but got None")
```

Comment:
Suggestion: if you move the `let zone_symbols = ...` above the `let mut time_zone_format = ...` then you don't have to do this because you haven't handed off ownership yet.

Reply:
@sffc 

I'm not quite clear on what you're suggesting here. 

Do you want us to clone the `patterns` argument?